org.sonatype.m2e.egit/src/org/sonatype/m2e/egit/internal/EgitScmHandler.java

```diff
@@ -101,6 +101,30 @@ protected String normalizeUri(String uri) throws URISyntaxException {
       throw new URISyntaxException(uri, "Invalid git URI");
     }
 
+    // Normalize uri like
+    // git@github.com:errai/errai.git/errai-common
+    // into
+    // git://github.com/errai/errai.git
+    // These 3 distinctions may happen independently in other cases.
+    String gitPrefix = "git://";
```

Comment:
Ok, let's first attempt connection with SSH, and if it fails, show message that suggests user connecting git anonymously. If user cancels, just do nothing, if user confirms, change URL from git@ into git:// and repeat the attempt.